lib/record_not_unique.rb

```diff
@@ -1,82 +1,100 @@
 # frozen_string_literal: true
 
 module RecordNotUnique
+  # This module provides methods for handling record not unique errors.
   module ClassMethods
+    # Handles the record not unique errors for the specified fields and error messages.
     def handle_record_not_unique(*args)
       # need to be shared across child classes as indexes would be common
       class_eval do
         cattr_accessor :_rnu_indexes, :_rnu_error_messages
-        
-        self._rnu_indexes = []
-        self._rnu_error_messages = []
+
+        initialize_rnu_attributes
         _indexes = connection.indexes(table_name)
         args.each do |arg|
-          raise NotImplementedError unless arg.key?(:field) && arg.key?(:message)
-          raise 'field should be an array' unless arg[:field].is_a?(Array)
-          _index = _indexes.detect { |index| index.columns.eql?(arg[:field]) }
-          raise "Couldn't find unique index for field #{arg[:field].join(',')}" if !_index&.unique
-          self._rnu_indexes << _index.name
-          self._rnu_error_messages << arg[:message].to_a.flatten
+          _rnu_indexes << _get_index_name(_indexes, arg)
+          _rnu_error_messages << Array(arg[:message]).flatten
         end
         prepend InstanceMethods
       end
     end
+
+    private
+
+    def initialize_rnu_attributes
+      self._rnu_indexes = []
+      self._rnu_error_messages = []
+    end
+
+    def _get_index(_indexes, field)
+      _indexes.detect { |index| index.columns.eql?(field) }
+    end
+
+    def _get_index_name(_indexes, arg)
+      raise ArgumentError, 'field and message must be provided' unless arg.key?(:field) && arg.key?(:message)
+      raise 'field should be an array' unless arg[:field].is_a?(Array)
+
+      _index = _get_index(_indexes, arg[:field])
+      raise "Couldn't find unique index for field #{arg[:field].join(',')}" unless _index&.unique
+
+      _index.name
+    end
   end
```

Comment:
all these methods will get extended in AR. Are they really needed? what's the rubocop issue here?

Reply:
Due to `CyclomaticComplexity`. it's above 30. ideally, it should be 17.